docs/user_guide/storing/index_in_memory.md

```diff
@@ -0,0 +1,224 @@
+# In-Memory Document Index
+
+
+
+[InMemoryDocIndex][docarray.index.backends.in_memory.InMemoryDocIndex] stores all Documents in DocLists in memory. 
+It is a great starting point for small datasets, but is not battle-tested in production.
```

Comment:
I am not sure if this caveat is needed here. I put it for the hnswlib implementation because i suspect that the combo between hnswlib and qlit can be flaky due to to data desync etc.